docs/design/design_trivy_file_system_scanner.md

```diff
@@ -0,0 +1,200 @@
+2. When image pulled from managed registry of managed cluster
+3. When pod is running with image present on host
+
+## Proposal
+
+Use Trivy file system scanning option to scan an image. So that we will not need to provide registry credentials to trivy.
+K8s scan job will use underline secrets available to pull an image and that image will be scanned by trivy as file system
+ scanning. Same idea is talked here in POC [#692](https://github.com/aquasecurity/starboard/discussions/692)
+
+### High level idea
+
+Run scan job with image which we want to scan and override the default entrypoint of image with following command
+```shell script
+trivy fs --ignore-unfixed /
+```
+With above command trivy will scan entire file system of an image and it will give us the result.
+
+To run trivy in that image we need trivy to be available in the container. To solve this problem we have a way for it.
+1. Add an init container with tvivy image in the scan job which will pull trivy image, Copy trivy executable in a volume
+ which will be shared with main container 
+3. Add another init container to download trivy database and mount db file in volume which will be used as trivy database
+during file system scan
+
+### TL;DR;
+
+Currently, clientServer mode is not supporting filesystem scan. So we will have to go with standalone mode. In future we
+ can change it to client server mode and accordingly we will have to change our scan job spec.
+ 
+ Configuration parameter can be added to enable this scan mode
+``trivy.command: fs``. This trivy.command will support two values ``image`` and ``fs``. For backward compatibility. if 
+this config parameter is not defined then it will consider it as ``image``.
+
+We have two options to consider here
+1. We can restrict scan job to run on same host where original pod is running.
+    - ImagePullPolicy can be set as false, so we will not need to pull the image on node
+    - nodeName has to be provided for scan job to run on specific node
+2. Allow scan job to run on any node in the cluster.  *This option will not be implemented for now*
+    - ImagePullPolicy should be set as IfNotPresent
+    - nodeName will be set as empty, so that scan job will run on any node.
+    - make ImagePullSecret available for scan job
+    
+## Scan job running on same host
+
+To run scan on same host where application pod is running then we have to identify host on which application pod is 
+running. To do this we will search for active pod for any controller and we will get the nodename from first pod of it.
+
+1. Scan job has to run on same host where pod is running.
+    - Get the nodename of one of the pod. If we dont get nodename then return error.
+    - No need to copy secret from the source namespace to starboard-operator namespace. We will ignore imagePullSecret of 
+    source application workload.
+    - ImagePullPolicy will be set as Never
+    - nodeName will be passed explicitly.
```

Comment:
Clarify that we set nodeName on a scan job so it's scheduled on the same node as scanned workload.